### PR TITLE
tests(ticdc): support for canal-json test and add basic case

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/4189
### What is changed and how it works?

* support for canal-json test
* add basic case

### Check List 

Tests 

 - Integration test

Code changes

None

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/hold

Waiting for https://github.com/PingCAP-QE/ci/pull/550 landed.

**Comment by hi-rustin** [comment]
Test result: https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_kafka_integration_test_copy/detail/cdc_ghpr_kafka_integration_test_copy/11/pipeline/134/

File: tests/integration_tests/canal_json_basic/data/data.sql [hunk 1]
```diff
@@ -0,0 +1,124 @@
+USE `test`;
```
Comment (3AceShowHand) [inline tests/integration_tests/canal_json_basic/data/data.sql:1]: It looks that this one can replace the case `CanalJson` in the GitHub action. 

Can we remove that one?

**Review by 3AceShowHand — APPROVED** [review]

File: tests/integration_tests/canal_json_basic/data/data.sql (hunk 1 above)
Comment (hi-rustin) [reply tests/integration_tests/canal_json_basic/data/data.sql:1]: Good catch! removed.

Thanks!

**Comment by hi-rustin** [comment]
/unhold 

https://github.com/PingCAP-QE/ci/pull/550 landed.

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**